### ISSUE TITLE
Flip animation is buggy on mobile when both `flipOnClick` and `flipOnHover` are `true`

### ISSUE THREAD

**Issue by omarryhan** [sbayd/react-flippy#9]
Hi,

Thanks for the very easy to use library.

I made a code sandbox that reproduces the error.
Here's the code: https://codesandbox.io/s/wispy-sun-vzz0d
And you can view it on your phone using this link: https://vzz0d.codesandbox.io/

**Comment by omarryhan** [comment]
Also, using this setup, it's breaking my website's right margin on mobile. It's adding a couple of extra pixels to the right.

**Comment by Apuello1990** [comment]
I have the same problem, any solution??

**Comment by velezhanski** [comment]
Bump

**Comment by york18** [comment]
Just to add more to @velezhanski's solution. If you want to "actively" check for mobile device (I needed this because I was testing my project on chrome desktop but with its mobile screen dev tool) then you can use a custom hook

Apparently `react-device-detect` does not check for mobile device dynamically (it only checks when the page is first loaded so if you switch from desktop to mobile or vice versa the `isMobile` value is not updated)

So, you can use this custom hook instead to check if its a mobile device:

```
import React, { useState, useEffect } from "react";
import Flippy, { FrontSide, BackSide } from "react-flippy";

export default function About() {
  const isMobile = useMobileDeviceCheck();
  return (
    <Flippy
      flipOnHover={isMobile ? false : true}
      flipOnClick={isMobile ? true : false}
      flipDirection="horizontal"
    >
      <FrontSide>
        This is front
      </FrontSide>

      <BackSide>
        This is back
      </BackSide>
    </Flippy>
  );
}

// Hook to detect mobile devices
function useMobileDeviceCheck() {
  const [isMobileDevice, setIsMobileDevice] = useState(false);

  useEffect(() => {
    // Handler to call on window resize
    function handleResize() {
      // Detect if current device is mobile
      setIsMobileDevice(!!navigator.maxTouchPoints);
    }

    // Add event listener
    window.addEventListener("resize", handleResize);

    // Call handler right away so state gets updated with initial device type
    handleResize();

    // Remove event listener on cleanup
    return () => window.removeEventListener("resize", handleResize);
  }, []); // Empty array ensures that effect is only run on mount

  return isMobileDevice;
}


```

**Comment by sbayd** [comment]
Hi,

Sorry for very late reply and thank you for your solution. 
This issue is fixed with v 1.1.0.

From now on I'll be maintaining the library much faster.
